diff -up guile-1.8.4/libguile/Makefile.in.multilib guile-1.8.4/libguile/Makefile.in
--- guile-1.8.4/libguile/Makefile.in.multilib	2008-02-20 19:34:05.000000000 +0100
+++ guile-1.8.4/libguile/Makefile.in	2008-02-20 19:34:05.000000000 +0100
@@ -1630,6 +1630,11 @@ install-nodist_modincludeHEADERS: $(nodi
 	  f=$(am__strip_dir) \
 	  echo " $(nodist_modincludeHEADERS_INSTALL) '$$d$$p' '$(DESTDIR)$(modincludedir)/$$f'"; \
 	  $(nodist_modincludeHEADERS_INSTALL) "$$d$$p" "$(DESTDIR)$(modincludedir)/$$f"; \
+	  if [ "$$f" = "scmconfig.h" ]; then \
+	    grep -q "SCM_SIZEOF_LONG 4" "$(DESTDIR)$(modincludedir)/$$f" && g=32 || g=64; \
+	    mv "$(DESTDIR)$(modincludedir)/$$f" "$(DESTDIR)$(modincludedir)/scmconfig-$$g.h"; \
+	    $(nodist_modincludeHEADERS_INSTALL) "$${d}scmconfig.h.mlib" "$(DESTDIR)$(modincludedir)/$$f"; \
+	  fi; \
 	done
 
 uninstall-nodist_modincludeHEADERS:
diff -up guile-1.8.4/libguile/guile-snarf.in.multilib guile-1.8.4/libguile/guile-snarf.in
--- guile-1.8.4/libguile/guile-snarf.in.multilib	2008-02-12 15:04:09.000000000 +0100
+++ guile-1.8.4/libguile/guile-snarf.in	2008-02-20 19:40:21.000000000 +0100
@@ -39,7 +39,6 @@
 #   #endif
 # 
 # If the environment variable CPP is set, use its value instead of the
-# C pre-processor determined at Guile configure-time: "@CPP@".
 
 # Code:
 
@@ -75,7 +74,7 @@ tempdir="$TMPDIR/guile-snarf.$$"
 (umask 077 && mkdir $tempdir) || exit 1
 temp="$tempdir/tmp"
 
-if [ x"$CPP" = x ] ; then cpp="@CPP@" ; else cpp="$CPP" ; fi
+if [ x"$CPP" = x ] ; then cpp="gcc -E" ; else cpp="$CPP" ; fi
 
 trap "rm -rf $tempdir" 0 1 2 15
 
diff -up /dev/null guile-1.8.4/libguile/scmconfig.h.mlib
--- /dev/null	2007-12-22 10:41:34.485021827 +0100
+++ guile-1.8.4/libguile/scmconfig.h.mlib	2008-02-20 19:34:05.000000000 +0100
@@ -0,0 +1,6 @@
+#include <bits/wordsize.h>
+#if __WORDSIZE == 32
+#include "libguile/scmconfig-32.h"
+#else
+#include "libguile/scmconfig-64.h"
+#endif
