commit 23eafdbac3bcb1116205a9ef51f0bd301c966483
Author: Tomas Korbar <tkorbar@redhat.com>
Date:   Mon Jan 25 10:04:19 2021 +0100

    Fix multilib

diff --git a/libguile/Makefile.am b/libguile/Makefile.am
index 74894e1..2ee2ef5 100644
--- a/libguile/Makefile.am
+++ b/libguile/Makefile.am
@@ -84,7 +84,10 @@ scmconfig.h: ${top_builddir}/config.h gen-scmconfig$(EXEEXT)
 	./gen-scmconfig$(EXEEXT) >> scmconfig.h.tmp; \
 	chmod 444 scmconfig.h.tmp; \
 	rm -f scmconfig.h; \
-	mv scmconfig.h.tmp scmconfig.h)
+	mv scmconfig.h.tmp scmconfig.h; \
+	grep -q "SCM_SIZEOF_LONG 4" ./scmconfig.h && g=32 || g=64; \
+	mv scmconfig.h scmconfig-$$g.h;\
+	cp scmconfig-multilib.h scmconfig.h)
 
 
 guile_filter_doc_snarfage_SOURCES = c-tokenize.c
@@ -661,7 +664,8 @@ modinclude_HEADERS =				\
 	vm-expand.h				\
 	vm.h					\
 	vports.h				\
-	weaks.h
+	weaks.h					\
+	scmconfig-[0-9][0-9].h
 
 nodist_modinclude_HEADERS = version.h scmconfig.h
 
diff --git a/libguile/scmconfig-multilib.h b/libguile/scmconfig-multilib.h
new file mode 100644
index 0000000..d7f96a2
--- /dev/null
+++ b/libguile/scmconfig-multilib.h
@@ -0,0 +1,7 @@
+/* multilib clean wrapper */
+#include <bits/wordsize.h>
+#if __WORDSIZE == 32
+#include "scmconfig-32.h"
+#else
+#include "scmconfig-64.h"
+#endif
diff --git a/meta/guile-config.in b/meta/guile-config.in
index 0226f68..5a6effc 100755
--- a/meta/guile-config.in
+++ b/meta/guile-config.in
@@ -1,7 +1,6 @@
 #!/bin/sh
-PKG_CONFIG_PATH="@pkgconfigdir@:$PKG_CONFIG_PATH"
 GUILE_AUTO_COMPILE=0
-export PKG_CONFIG_PATH GUILE_AUTO_COMPILE
+export GUILE_AUTO_COMPILE
 
 exec "@installed_guile@" -e main -s $0 "$@"
 !#
@@ -31,7 +30,7 @@ exec "@installed_guile@" -e main -s $0 "$@"
              (ice-9 rdelim))
 
 
-(define %pkg-config-program "@PKG_CONFIG@")
+(define %pkg-config-program "pkg-config")
 
 ;;;; main function, command-line processing
 
