diff -up guile-1.8.5/configure.in.macrodir guile-1.8.5/configure.in
--- guile-1.8.5/configure.in.macrodir	2008-05-07 19:51:15.000000000 +0200
+++ guile-1.8.5/configure.in	2008-11-19 13:58:34.000000000 +0100
@@ -38,7 +38,7 @@ AC_INIT(patsubst(m4_esyscmd(. ./GUILE-VE
 ]),
         [bug-guile@gnu.org])
 AC_CONFIG_AUX_DIR([build-aux])
-AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_MACRO_DIR([guile-config])
 AC_CONFIG_SRCDIR(GUILE-VERSION)
 
 AM_INIT_AUTOMAKE([gnu no-define check-news])
